internal/resources/connections/data_source_metrics_endpoint_scrape_job.go

```diff
@@ -0,0 +1,111 @@
+package connections
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common"
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common/connectionsapi"
+)
+
+type datasourceMetricsEndpointScrapeJob struct {
+	client *connectionsapi.Client
+}
+
+func makeDatasourceMetricsEndpointScrapeJob() *common.DataSource {
+	return common.NewDataSource(
+		common.CategoryConnections,
+		resourceMetricsEndpointScrapeJobTerraformName,
+		&datasourceMetricsEndpointScrapeJob{},
+	)
+}
+
+func (r *datasourceMetricsEndpointScrapeJob) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	// Configure is called multiple times (sometimes when ProviderData is not yet available), we only want to configure once
+	if req.ProviderData == nil || r.client != nil {
+		return
+	}
+
+	client, err := withClientForDataSource(req, resp)
+	if err != nil {
+		return
+	}
+
+	r.client = client
+}
+
+func (r *datasourceMetricsEndpointScrapeJob) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = resourceMetricsEndpointScrapeJobTerraformName
+}
+
+func (r *datasourceMetricsEndpointScrapeJob) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The Terraform Resource ID. This has the format \"{{ stack_id }}:{{ name }}\".",
+				Computed:    true,
+			},
+			"stack_id": schema.StringAttribute{
+				Description: "The Stack ID of the Grafana Cloud instance. Part of the Terraform Resource ID.",
+				Required:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the Metrics Endpoint Scrape Job. Part of the Terraform Resource ID.",
+				Required:    true,
+			},
+			"enabled": schema.BoolAttribute{
+				Description: "Whether the metrics endpoint scrape job is enabled or not.",
+				Computed:    true,
+			},
+			"authentication_method": schema.StringAttribute{
+				Description: "Method to pass authentication credentials: basic or bearer.",
+				Computed:    true,
+			},
+			"authentication_bearer_token": schema.StringAttribute{
```

Comment:
We're not reading these are we? Does it make sense for them to be included in the schema?